### ISSUE TITLE
Year selector calling "shouldDisableDate"

### ISSUE THREAD

**Issue by mauriciocarnieletto** [mui-org/material-ui-pickers#1899]
## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers |    v4-0-0-alpha-8     |
| material-ui          |    ^4.10.1     |
| TypeScript           |   ^3.8.3      |
| React                |    "^16.13.1"     |
| Browser              |     any    |
| Peer library         |    date-fns     |  |

## Steps to reproduce

1. 2. 3.
1. Create a custom function to disable weekends, for example
```
  function shouldDisableDate(day, pickerProps) {
    return day.getDay() === 0 || day.getDay() === 6;
  }

```
And call it in any datepicker.

This function is being called at the year selector, disabling years that should not be disabled

![image](https://user-images.githubusercontent.com/4827003/84700667-f9f00c80-af29-11ea-855d-004c64b36fa4.png)

## Expected behavior

Year selector screen should only call shouldDisableYear right?

## Actual behavior

![image](https://user-images.githubusercontent.com/4827003/84700754-23109d00-af2a-11ea-8467-9e78ed8fc5ed.png)

https://codesandbox.io/s/cranky-black-iuqu4?file=/src/App.jsx:263-368

Couldn't get it running, but you can see the code

**Comment by oliviertassinari** [comment]
Related to #1774 and #1002.

**Comment by dmtrKovalenko** [comment]
In short use `shouldDisableYear`

**Comment by mauriciocarnieletto** [comment]
I don't know if I was clear, in short, the year picker is calling shouldDisableDate. It shouldn't right?

It should only take shouldDisableYear wright?

Sorry about my english :T

**Comment by oliviertassinari** [comment]
@mauriciocarnieletto Yes, I would only expect `shouldDisableYear` to be called in your context.

**Comment by oliviertassinari** [comment]
Here is an updated reproduction that run: https://codesandbox.io/s/nervous-drake-85kb1?file=/src/App.jsx.

**Comment by dmtrKovalenko** [comment]
This bug appeared by the fix of #1774. Going to revert changes from #1848
